ui/app/adapters/job.js

```diff
@@ -159,4 +166,98 @@ export default class JobAdapter extends WatchableNamespaceIDs {
       },
     });
   }
+
+  runAction(job, action, allocID) {
+    let messageBuffer = '';
+
+    const protocol = window.location.protocol === 'https:' ? 'wss:' : 'ws:';
+    const region = this.system.activeRegion;
+    const applicationAdapter = getOwner(this).lookup('adapter:application');
+    const prefix = `${
+      applicationAdapter.host || window.location.host
+    }/${applicationAdapter.urlPrefix()}`;
+
+    const wsUrl =
+      `${protocol}//${prefix}/job/${job.get('id')}/action` +
```

Comment:
```suggestion
      `${protocol}//${prefix}/job/${encodeURIComponent(job.get('id')(}/action` +
```

Not sure if this is handled at another level, but job IDs can have pretty much any character 😬